typesocket/typesocket_test.go

```diff
@@ -73,3 +73,43 @@ func TestTypeSocket(t *testing.T) {
 		t.Fatal("timeout")
 	}
 }
+
+func TestLatency(t *testing.T) {
+	serverPublic, serverPrivate, _ := ed25519.GenerateKey(rand.Reader)
+
+	serverMux := NewMux(map[string]interface{}{})
+	hub := &Hub{
+		Mux:   serverMux,
+		conns: make(map[*serverConn]bool),
+	}
+	httpMux := http.NewServeMux()
+	httpMux.Handle("/ws", hub)
+	l, err := edtls.Listen("tcp", "127.0.0.1:0", serverPrivate)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer l.Close()
+	go http.Serve(l, httpMux)
+
+	clientMux := NewMux(map[string]interface{}{})
+
+	conn, err := Dial(fmt.Sprintf("wss://%s/ws", l.Addr().String()), serverPublic)
+	if err != nil {
+		t.Fatal(err)
+	}
+	go conn.StartPings(10 * time.Millisecond)
+	go conn.Serve(clientMux)
+	defer conn.Close()
+
+	time.Sleep(10 * time.Millisecond)
+
+	latency := conn.Latency()
+	if latency != 0 {
+		t.Fatalf("latency should be 0, was %s", latency)
+	}
+
+	time.Sleep(5 * time.Millisecond)
+	if conn.Latency() <= 0 {
+		t.Fatal("should have positive latency")
```

Comment:
Why do you expect the latency to be positive here but zero above?

Reply:
The latency should be zero until the ping period has elapsed (10ms here). Not sure if the check for 0 is useful to include in the test.

Reply:
I'd remove the 0 check.

Reply:
Removed